From e0d31108799303339b58c3ee66045058fe253461 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Mon, 16 Jul 2018 08:13:40 -0500
Subject: [PATCH 4325/5242] MLK-18904-3: imx: hdp: Moving
 AFE_check_rate_supported, adding support for
 i.MX8M (MCU2) DP PHY

commit  0c2ad2a620d340f7e2bad83cf38d60602488c209 from
https://source.codeaurora.org/external/imx/linux-imx.git

Moving AFE_check_rate_supported which is used by i.MX8QM and i.MX8M
Adding headers for i.MX8M DP PHY.

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp/API_AFE.c |   19 ++++++++++++++++++-
 drivers/mxc/hdp/API_AFE.h |    4 ++++
 2 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/hdp/API_AFE.c b/drivers/mxc/hdp/API_AFE.c
index f4ce67b..060a6a49 100644
--- a/drivers/mxc/hdp/API_AFE.c
+++ b/drivers/mxc/hdp/API_AFE.c
@@ -35,7 +35,7 @@
  * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  *
- * Copyright 2017 NXP
+ * Copyright 2017-2018 NXP
  *
  ******************************************************************************
  *
@@ -169,3 +169,20 @@ int get_table_row(
 	}
 	return i;
 }
+
+u8 AFE_check_rate_supported(ENUM_AFE_LINK_RATE rate)
+{
+	switch (rate) {
+	case AFE_LINK_RATE_1_6:
+	case AFE_LINK_RATE_2_1:
+	case AFE_LINK_RATE_2_4:
+	case AFE_LINK_RATE_2_7:
+	case AFE_LINK_RATE_3_2:
+	case AFE_LINK_RATE_4_3:
+	case AFE_LINK_RATE_5_4:
+		return 1;
+	default:
+		return 0;
+	}
+}
+
diff --git a/drivers/mxc/hdp/API_AFE.h b/drivers/mxc/hdp/API_AFE.h
index 91f9c5a..37ddaa4 100644
--- a/drivers/mxc/hdp/API_AFE.h
+++ b/drivers/mxc/hdp/API_AFE.h
@@ -91,8 +91,12 @@
 u16 Afe_read(state_struct *state, u32 offset);
 void AFE_init(state_struct *state, int num_lanes,
 	      ENUM_AFE_LINK_RATE link_rate);
+void afe_init_t28hpc(state_struct *state, int num_lanes,
+	      ENUM_AFE_LINK_RATE link_rate);
 void AFE_power(state_struct *state, int num_lanes,
 	       ENUM_AFE_LINK_RATE link_rate);
+void afe_power_t28hpc(state_struct *state, int num_lanes,
+	       ENUM_AFE_LINK_RATE link_rate);
 void set_field_value(reg_field_t *reg_field, u32 value);
 int set_reg_value(reg_field_t reg_field);
 int inside(u32 value, u32 left_sharp_corner, u32 right_sharp_corner);
-- 
1.7.9.5

